chirho/observational/handlers/soft_conditioning.py

```diff
@@ -190,3 +203,127 @@ def configure(
         raise NotImplementedError(
             f"Could not reparameterize deterministic site {msg['name']}"
         )
+
+
```

Comment:
Let's move all of the new code below into a separate file `handlers/cut.py`

Reply:
Good idea! I moved this code to cut.py.